### PR TITLE
Implement recursive fetching for sites

### PR DESCRIPTION
**Pull request by tienhoah**

### CODE REVIEW

File: src/components/MermaidDash.js [hunk 1]
```diff
@@ -244,6 +240,27 @@ class MermaidDash extends Component {
...
+  };
+
+  fetchEntiresSites = async (params, pageNo = 1) => {
+    const results = await this.fetchSitesChunk(params, pageNo);
+    if (results.length === this.state.queryLimit) {
+      return [...results].concat(await this.fetchEntiresSites(params, pageNo + 1));
+    } else {
+      return results;
+    }
+  };
```
Comment (brianbancroft) [inline src/components/MermaidDash.js:262]: I'm curious about how this particular bit of recursion is implemented. I'm looking at line `257` where you're doing another call only if `results.length === this.state.queryLimit`.

Can you explain what happens if the number of results in the last page coincides with the limit? Will another request happen, and if so, what happens if you fetch another query with no results? 

Ideally, the best practice is to minimize api calls. Perhaps there's an opportunity to jam in a "total records" parameter in the API response that'll help out here...

File: src/components/MermaidDash.js [hunk 2]
```diff
@@ -244,6 +240,27 @@ class MermaidDash extends Component {
...
+      }
+    });
+    siteResults = apiResults;
+    return Object.keys(siteResults).length === 0 ? [] : siteResults.data.features;
+  };
+
+  fetchEntiresSites = async (params, pageNo = 1) => {
+    const results = await this.fetchSitesChunk(params, pageNo);
+    if (results.length === this.state.queryLimit) {
+      return [...results].concat(await this.fetchEntiresSites(params, pageNo + 1));
```
Comment (brianbancroft) [inline src/components/MermaidDash.js:258]: I find it interesting that you're using both concat and spread syntax at the same time.

**Comment by saanobhaai** [comment]
Thanks @brianbancroft -- Nick note there is in fact a `count` returned that you could use.

Best,
Kim

[WCS logo]

Kim Fisher
Spatial Analyst and Developer
Wildlife Conservation Society
kfisher@wcs.org<mailto:kfisher@wcs.org>
welikia.org<https://welikia.org/> | visionmaker.nyc<https://visionmaker.nyc/> | datamermaid.org<https://datamermaid.org/>
skype: wcs.org_kfisher | calendar<https://calendly.com/kimfisher>



________________________________

File: src/components/Header.js [hunk 3]
```diff
@@ -61,7 +61,7 @@ const HeaderItems = headerContent.map(({ name, link }) => {
 
 const Header = () => {
   const classes = headerStyles();
-  const [open, setModalStage] = useState(false);
+  const [open, setModalStage] = useState(true);
```
Comment (brianbancroft) [inline src/components/Header.js:64]: Would suggest a different pair of names. The convention is [a, setA] 

in this case, would it be modalStageOpen?

**Comment by tienhoah** [comment]
closes #177

**Pull request merged**